static/app/views/starfish/queries/useProjectSpanMetricsCounts.tsx

```diff
@@ -5,31 +5,33 @@ import {useSpansQuery} from 'sentry/views/starfish/utils/useSpansQuery';
 interface Options {
   enabled?: boolean;
   projectId?: string[];
+  query?: string;
+  statsPeriod?: string;
 }
 
-export const useHasAnySpanMetrics = ({projectId, enabled}: Options) => {
+export const useProjectSpanMetricCounts = ({
+  projectId,
+  enabled,
+  query,
+  statsPeriod,
+}: Options) => {
   const eventView = EventView.fromSavedQuery({
     name: 'Has Any Span Metrics',
-    query: '',
-    fields: ['count()'],
+    query,
+    fields: ['project.id', 'count()'],
     projects: projectId && projectId.map(id => parseInt(id, 10)),
     dataset: DiscoverDatasets.SPANS_METRICS,
     version: 2,
   });
 
-  eventView.statsPeriod = SAMPLE_STATS_PERIOD;
+  eventView.statsPeriod = statsPeriod;
```

Comment:
nit: Couldn't you set this in `fromSavedQuery` too?

Reply:
I assumed so, but the typing made it seem like I can't for some reason!